corehq/apps/app_manager/fields.py

```diff
@@ -128,40 +143,69 @@ def _add_choices(field, choices):
             for app_data in self.all_sources.values():
                 app_data['data_source'] = [{"text": ds.display_name, "value": ds.data_source_id}
                                            for ds in available_data_sources]
+        self.registry_slug_field.choices = sort_tuple_field_choices_by_name(
+            [(registry["slug"], registry["name"]) for registry in get_data_registry_dropdown_options(domain)],
+        ) + [('', '--------')]
 
         # NOTE: This corresponds to a view-model that must be initialized in your template.
         # See the doc string of this class for more information.
         self.application_field.widget.attrs = {'data-bind': 'value: application'}
+        self.registry_slug_field.widget.attrs = {'data-bind': '''
+            optionsText: function(item){return item.text},
+            optionsValue: function(item){return item.value},
+            value: registrySlug
+            '''}
         self.source_type_field.widget.attrs = {'data-bind': 'value: sourceType'}
         self.source_field.widget.attrs = {'data-bind': '''
-            options: sourcesMap[application()][sourceType()],
             optionsText: function(item){return item.text},
-            optionsValue: function(item){return item.value}
+            optionsValue: function(item){return item.value},
+            value: sourceId
         '''}
 
+        if self.enable_registry:
+            self.application_field.widget.attrs['data-bind'] += ", disable:" \
+                                                                "isDataFromOneProject() != 'true'," \
+                                                                "optionsText: function(item){return item.text}," \
+                                                                "optionsValue: function(item){return item.value}," \
+                                                                "options: dropdownMap['app'][isDataFromOneProject()]"
```

Comment:
nit: reformat
```suggestion
            self.application_field.widget.attrs['data-bind'] += (
                ", disable:"
                "isDataFromOneProject() != 'true',"
                "optionsText: function(item){return item.text},"
                "optionsValue: function(item){return item.value},"
                "options: dropdownMap['app'][isDataFromOneProject()]"
            )
```

Also applies to the other lines